### PR TITLE
Add room repo

### PR DESCRIPTION
**Pull request by viktor533**

### CODE REVIEW

File: src/main/java/com/epam/repo/RoomRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+        @Cleanup
+        PreparedStatement statement = getPreparedStatement(findSQLRequest);
+
+        try {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                room = Room.builder()
+                        .hotelId(resultSet.getLong("HOTEL_ID"))
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/RoomRepositoryImpl.java]: Extract column names

they are always the same

File: src/main/java/com/epam/repo/RoomRepositoryImpl.java [hunk 2]
```diff
@@ -21,6 +21,11 @@
     private static final String updateSQLRequest = "UPDATE ROOM SET ID \'?\', HOTEL_ID \'?\', NUM_OF_GUESTS = \'?\', PRICE_PER_NIGHT \'?\', CLASS = \'?\' WHERE ID = ?;";
     private static final String findAllSQLRequest = "SELECT * FROM HOTEL;";
 
+    private static final String hotelIdColumnName = "HOTEL_ID";
+    private static final String numOfGuestsColumnName = "NUM_OF_GUESTS";
+    private static final String pricePerNightColumnName = "PRICE_PER_NIGHT";
```
Comment (IlyasYOY) [inline src/main/java/com/epam/repo/RoomRepositoryImpl.java]: Naming conventions: 

consts are named like HOTEL_ID_COLUMN_NAME

File: src/main/java/com/epam/repo/RoomRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        if (room == null) {
+            return null;
+        }
+
+        @Cleanup
+        PreparedStatement statement = getPreparedStatement(SAVE_SQL_REQUEST);
+        setRoomToPreparedStatement(room, statement);
+        statement.execute();
+
+//        TODO: bookingRepositoryImp
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/repo/RoomRepositoryImpl.java]: you can resolve this TODO now, we are already have this repo

Comment (imbirevkotlya) [reply src/main/java/com/epam/repo/RoomRepositoryImpl.java]: And all blocks from the code below

Comment (viktor533) [reply src/main/java/com/epam/repo/RoomRepositoryImpl.java]: already done

File: src/main/java/com/epam/repo/RoomRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        }
+
+        return room;
+    }
+
+    @Override
+    @SneakyThrows
+    public Room update(Room room) {
+        if (room == null) {
+            return null;
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/repo/RoomRepositoryImpl.java]: Why you want to return null here, and to use exceptions?

Comment (imbirevkotlya) [reply src/main/java/com/epam/repo/RoomRepositoryImpl.java]: And in all other cases?

**Pull request merged**